src/clients/detector.rs

```diff
@@ -65,6 +66,24 @@ impl DetectorClient {
             .await?;
         Ok(response)
     }
+
+    pub async fn generation_detection(
+        &self,
+        model_id: &str,
+        request: GenerationDetectionRequest,
+    ) -> Result<Vec<GenerationDetectionResponse>, Error> {
+        let client = self.client(model_id)?;
+        let url = client.base_url().as_str();
+        let response = client
+            .post(url)
+            .header(DETECTOR_ID_HEADER_NAME, model_id)
+            .json(&request)
+            .send()
+            .await?
+            .json()
+            .await?;
+        Ok(response)
+    }
```

Comment:
This method is literally a copy of text_contents, but with different types for the `request` parameter and the return type. I tried applying generics here, but I couldn't make it work.

Reply:
@evaline-ju Should I open up a story for this, so we don't have this PR hanging for too long?